### PR TITLE
chore: run CI on all platforms

### PR DESCRIPTION
**Pull request by dsanders11**
Run CI on all the platforms... except MacOS for tests.

Couldn't get the MacOS tests to pass on CI, if anyone else wants to take a look.

### CODE REVIEW

**Comment by codebytere** [comment]
lol https://github.com/electron/build-tools/pull/206#issuecomment-713221204

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -5,9 +5,28 @@ on: pull_request
 jobs:
   build:
     name: Test pull requests
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (codebytere) [inline .github/workflows/pull_request.yml:10]: think this is missing `macos-latest`?

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -5,9 +5,28 @@ on: pull_request
...
+    - name: Set git global config (Windows)
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: |
+        git config --global core.autocrlf false
+        git config --global core.eol lf
+        git config --global core.filemode false
+        git config --global branch.autosetuprebase always
+    - name: Install dependencies (Windows)
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: |
```
Comment (codebytere) [inline .github/workflows/pull_request.yml:23]: since this is one line we can just do `run: pip install pywin32`

same below

**Comment by codebytere** [comment]
Merged https://github.com/electron/build-tools/pull/206 - should be good to rebase

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (dsanders11) [reply .github/workflows/pull_request.yml:10]: Tests won't pass for some reason, it doesn't find files in the sandboxed directories it seems. So I left it off for now until someone can debug.

Comment (codebytere) [reply .github/workflows/pull_request.yml:10]: weird - in that case should remove the macos-specific bits farther down then probably

Comment (dsanders11) [reply .github/workflows/pull_request.yml:10]: Removed, and sure.

File: .github/workflows/pull_request.yml (hunk 2 above)
Comment (dsanders11) [reply .github/workflows/pull_request.yml]: Changed.

**Comment by dsanders11** [comment]
@codebytere, rebased and good to go.

**Comment by dsanders11** [comment]
Hmm, after the CI builds finished it was hung waiting on status to be reported. Found some info saying it might be related to `actions/checkout@v2`, which I had bumped to with this change. Changing back to `actions/checkout@v1` didn't fix it.

[This post](https://github.community/t/github-actions-pull-requests-always-include-some-outdated-checks/16157) suggests that possibly the 'Required' check is on the old non-matrix version of the action and so it won't ever return unless manually fixed?

**Comment by codebytere** [comment]
@dsanders11 i fixed that up - it's a branch protection rule and github hasn't quite smoothed out their logic to detect when a required check no longer exists

**Pull request merged**